### ISSUE TITLE
Direction parameter: magic constants

### ISSUE THREAD

**Issue by HonoluluHenk** [Teun/thenBy.js#34]
What always strikes me as a bit annoying are the magic numbers for 1/-1 for up/down.
Most programming docs refer to sorting directions as "ascending" or "descending or asc/desc for short.

Might be cool to have direction/options support this as well, resulting in something like:

type SortOrder = -1 | 1 | 'asc' | 'desc';
export function firstBy<T>(key: (v1: T, v2: T) => any), direction?: SortOrder | opt): IThenBy<T>;

**Comment by Teun** [comment]
While I think your naming (SortOrder) is better, I think accepting 'asc' and 'desc' as valid values would not be a *.d.ts-only change. Maybe make it an enum? So on the javascript level still work with -1, but on the typescript level, you could pass SortOrder.Asc?

Or was you suggestion intended for improving the JS side too?

**Comment by HonoluluHenk** [comment]
> Or was you suggestion intended for improving the JS side too?

yes :)

It'd enable me to write more beautiful/readable code while still being typesafe:
`foo.sort(firstBy('lastName', 'desc'))`
instead of (works but feels somehow clunky):
`foo.sort(firstBy('lastName', SortOrder.DESC))`

**Comment by Teun** [comment]
I don't understand why something with quotes around it is nicer/more beautiful than having an enum for that. It seems closer to the "magic" -1. Now it is a magic 'desc'.

My priority is:
1. keep backward compatibility for people already passing -1.
2. Make the typing layer discoverable and helpful in TypeScript

**Comment by HonoluluHenk** [comment]
Hmm... asc/desc are (imho) canonical abbreviations for ascending/descending thus giving quite a lot of meaning... so asc/desc are not magic constants.
Btw.: even [lodash.orderBy](https://lodash.com/docs/4.17.15#orderBy) is using this convention!

At least in Typescript, typedefs are also typesafe:
Typedefs are not just 'something with quotes around it': the compiler actually checks for allowed values in your code!
And IDEs let me code-complete *based on this type* (will probably also work in javascript if your IDE understands typings)

API backwards compatibility could be achieved like so:
`type SortOrder = 'asc' | 1 | 'desc' | -1;`
and
`if(opt.direction === -1 || opt.direction === 'desc') return function(v1,v2){return -f(v1,v2)};`

Maybe bump to a [new major version](https://semver.org/) if you deem this too hot?

But I let this wander off to a petty discussion about API styles...
Anyway, an enum is waaaaaaay better than 1/-1 so don't let me slow you down anymore :)

**Comment by Teun** [comment]
OK, I gave this some more thought. I think, if this were intended as a pure TypeScript solution, I would prefer an enum. Like SortOrder.Desc. However, I want thenby to be a player in the vanilla javascript world too. And I agree that -1 is much uglier than 'desc'. I would also like code samples to be identical for javscript and TypeScript as much as possible. 

So I decided to not have an enum, but rather define a type that combines all valid values. For the typing pull request, I will only change the type definition and make it 1 | -1. Later I will include this proposal too, implementing the value 'desc' in the library code and in the type definition.
